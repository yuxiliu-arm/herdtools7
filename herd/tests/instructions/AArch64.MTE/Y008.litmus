AArch64 Y008
(* IRG GCR_EL1 RRND=0, RGSR_EL1 tag and seed update *)
{
int x=100;
0:X0=x;
0:GCR_EL1=0;
0:RGSR_EL1=0x001400; (* RGSR_EL1 SEED=0x0014 TAG=0 *)
}
 P0         ;
 IRG X1,X0  ; (* After IRG, RGSR_EL1 TAG=0b1100, SEED=0x3001 *)
forall (0:X1=x:t12 /\ 0:RGSR_EL1=0x30010C)

(* Next seed calculation: *)
(* SEED before | lfsr bits       | top | SEED after *)
(* 0x0014      | 0 _1 0 1 _0 0   | 1   | 0x800A     *)
(* 0x800A      | 0 _0 1 0 _1 0   | 1   | 0xC005     *)
(* 0xC005      | 0 _0 0 1 _0 1   | 0   | 0x6003     *)
(* 0x6003      | 0 _0 0 0 _1 0   | 0   | 0x3001     *)

