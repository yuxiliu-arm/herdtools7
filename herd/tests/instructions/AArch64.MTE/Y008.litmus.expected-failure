Contradiction on reg 0:RGSR_EL1: loaded 5120 vs. stored 655360

Fatal: File "./herd/tests/instructions/AArch64.MTE/Y008.litmus" Adios
Fatal error: exception File "herd/mem.ml", line 841, characters 16-22: Assertion failed
