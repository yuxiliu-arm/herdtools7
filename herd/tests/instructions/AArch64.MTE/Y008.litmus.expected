Test Y008 Required
States 1
0:X1=x:t0; 0:RGSR_EL1=163840;
No
Witnesses
Positive: 0 Negative: 1
Condition forall (0:X1=x:t10 /\ 0:RGSR_EL1=5242890)
Observation Y008 Never 0 1
Hash=fbb85f77dfa0c46db6367f09d5c77bbd

